comp/core/gui/guiimpl/systray/Sources/WiFiIPCServer.swift

```diff
@@ -0,0 +1,249 @@
+import Foundation
+
+/// Request structure for IPC commands
+struct WiFiIPCRequest: Codable {
+    let command: String
+}
+
+/// Response structure for IPC
+struct WiFiIPCResponse: Codable {
+    let success: Bool
+    let data: WiFiData?
+    let error: String?
+}
+
+/// WiFiIPCServer handles Unix socket communication for WiFi data
+class WiFiIPCServer {
+    private let wifiDataProvider: WiFiDataProvider
+    private var socketFileDescriptor: Int32 = -1
+    private var isRunning = false
+    private var acceptQueue: DispatchQueue
+    private let socketPath: String
+
+    init(wifiDataProvider: WiFiDataProvider) {
+        self.wifiDataProvider = wifiDataProvider
+
+        // Get installation path from environment (set by LaunchAgent) or use default
+        let installPath = ProcessInfo.processInfo.environment["DD_INSTALL_PATH"] ?? "/opt/datadog-agent"
+
+        // Create socket path based on current user's UID
+        let uid = getuid()
+        self.socketPath = "\(installPath)/run/gui-\(uid).sock"
```

Comment:
Is install path or any path for that matter really required to create a socket. We can pick any unique name plus uid and call it a day?